src/marqo/core/unstructured_vespa_index/unstructured_document.py

```diff
@@ -93,6 +95,17 @@ def from_marqo_document(cls, document: Dict, filter_string_max_length: int) -> "
             elif isinstance(value, float):
                 instance.fields.float_fields[key] = value
                 instance.fields.score_modifiers_fields[key] = value
+            elif isinstance(value, dict) and not marqo_index_version_lt_2_9_0:
+                for k, v in value.items():
+                    if isinstance(v, int):
+                        instance.fields.int_fields[f"{key}.{k}"] = v
+                        instance.fields.score_modifiers_fields[f"{key}.{k}"] = v
+                    elif isinstance(v, float):
+                        instance.fields.float_fields[f"{key}.{k}"] = float(v)
+                        instance.fields.score_modifiers_fields[f"{key}.{k}"] = v
+            elif isinstance(value, dict) and marqo_index_version_lt_2_9_0:
+                raise UnsupportedFeatureError(f"Document {document} with a map field {key} "
```

Comment:
Not a good idea to print out the document. How about "Field {key} is a map field. Only indexes created with Marqo 2.9 or later support map fields". Please change the other exception below as well to avoid printing out the document.

Please also make sure this makes the individual document fail, not the full batch (i.e. cover this with a test if it isn't covered).